src/main/java/emissary/place/RefreshableServiceProviderPlace.java

```diff
@@ -0,0 +1,126 @@
+package emissary.place;
+
+import emissary.config.ConfigUtil;
+import emissary.config.Configurator;
+
+import org.apache.commons.collections4.CollectionUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.Nullable;
+
+/**
+ *
+ */
+public abstract class RefreshableServiceProviderPlace extends ServiceProviderPlace {
+
+    private final AtomicBoolean invalid = new AtomicBoolean(false);
+
+    public RefreshableServiceProviderPlace() throws IOException {}
+
+    public RefreshableServiceProviderPlace(final String thePlaceLocation) throws IOException {
+        super(thePlaceLocation);
+    }
+
+    protected RefreshableServiceProviderPlace(final String configFile, @Nullable final String theDir, final String thePlaceLocation)
+            throws IOException {
+        super(configFile, theDir, thePlaceLocation);
+    }
+
+    protected RefreshableServiceProviderPlace(final InputStream configStream, @Nullable final String theDir, final String thePlaceLocation)
+            throws IOException {
+        super(configStream, theDir, thePlaceLocation);
+    }
+
+    protected RefreshableServiceProviderPlace(final InputStream configStream) throws IOException {
+        super(configStream);
+    }
+
+    protected RefreshableServiceProviderPlace(final String configFile, final String placeLocation) throws IOException {
+        super(configFile, placeLocation);
+    }
+
+    protected RefreshableServiceProviderPlace(final InputStream configStream, final String placeLocation) throws IOException {
+        super(configStream, placeLocation);
+    }
+
+    public boolean isInvalid() {
+        return this.invalid.get();
+    }
+
+    public void invalidate() {
+        this.invalid.set(true);
+    }
+
+    /**
+     * Reinitialize the place by reloading the configurator and reconfiguring the place
+     */
+    public synchronized void refresh() {
+        try {
+            if (this.invalid.get()) {
+                this.configG = reloadConfigurator();
+                reconfigurePlace();
+            }
```

Comment:
Might even want this bumped up to ERROR level
```suggestion
            } else {
                logger.warn("cannot refresh place configuration without first calling invalidate; no reconfiguration performed");
            }
```